clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1142,6 +1156,7 @@ Node selectReadReplica(TopicPartition partition, Node leaderReplica, long curren
 
             Optional<Node> leaderOpt = position.currentLeader.leader;
             if (!leaderOpt.isPresent()) {
+                log.info("Requesting metadata update for partition {} since the position {} is missing the current leader node", partition, position);
```

Comment:
This feels more like a debug level log to me. Was there a specific reason we needed it?